view/frontend/web/js/view/payment/method-renderer/gr4vy-method.js

```diff
@@ -39,13 +39,14 @@ define(
                 var This = this;
                 storage.post( serviceUrl, JSON.stringify(payload)).done(
                     function (response) {
-                        var embed_token = response[0];
-                        var amount = response[1];
-                        var buyer_id = response[2];
-                        var cartItems = response[3];
+                        var total_mismatch = response[0];
+                        var embed_token = response[1];
+                        var amount = response[2];
+                        var buyer_id = response[3];
+                        var cartItems = response[4];
 
                         // Verify data before setting gr4vy
-                        if (embed_token && amount && buyer_id) {
+                        if (!total_mismatch && embed_token && amount && buyer_id) {
```

Comment:
@RowanBurgess if the total amount and cart item totals do not match, we still want to send the transaction but without sending cartItems in the setup function below.